### PR TITLE
contrib: review dev env and demo project setup

### PR DESCRIPTION
**Pull request by jorgeorpinel**
> Possible follow-up: https://github.com/iterative/vscode-dvc/pull/1928#issuecomment-1171595593

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -55,17 +55,16 @@ encouraged to try other DVC repositories -- especially real-world cases!
...
+  setting up the project.
 
-- `yarn install` is a prerequisite for automatically setting up the demo
-  project.
+- Go back to the root of this repo (not `./demo`) and `yarn setup:venv` to setup
+  the Python virtual environment in the demo project.
 
-- Run `yarn setup:venv` from the monorepo root (not `./demo`) to automatically
-  setup the Python virtual environment in the demo project.
+  > Python 3.8 is recommended, as some dependencies are not yet available for
```
Comment (shcheklein) [inline CONTRIBUTING.md:64]: hmm. I'm using Python 3.9 I think

Comment (jorgeorpinel) [reply CONTRIBUTING.md:64]: Same and so far no issues. This info was there already so I'm assuming it's correct.

Comment (shcheklein) [reply CONTRIBUTING.md:64]: got it , I think we can remove it

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -55,17 +55,16 @@ encouraged to try other DVC repositories -- especially real-world cases!
...
+  > Python 3.9.
 
-- Once the environment has been setup you should navigate into the demo folder
-  activate the virtual environment and run `dvc pull`.
+- Return to `demo/`, activate the virtual environment, and run `dvc pull`.
```
Comment (jorgeorpinel) [inline CONTRIBUTING.md:67]: How do I activate the env? 😅 I'm not sure what kind of env it is even. There's no `activate` binary. I see:

```console
$ ls .env/bin 
__pycache__            dul-upload-pack        ftfy                   pip3                   python3.9              tqdm
cmark                  dulwich                futurize               pip3.10                shortuuid              wheel
convert-caffe2-to-onnx dvc                    jp.py                  pip3.9                 shtab
convert-onnx-to-caffe2 f2py                   normalizer             pygmentize             slugify
distro                 f2py3                  pasteurize             python                 tabulate
dul-receive-pack       f2py3.9                pip                    python3                torchrun
```

@iterative/vs-code 🙏🏼

File: CONTRIBUTING.md (hunk 1 above)
Comment (jorgeorpinel) [reply CONTRIBUTING.md]: Gone!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -47,47 +47,48 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
...
+try other DVC repositories -- especially real-world cases!
 
-- Go to the `demo/` directory of this repo.
+- Install [Yarn](https://yarnpkg.com/) if needed. Go to the `demo/` directory of
+  this repo (`cd demo`) and run `yarn install` to set up the project.
 
-- It is recommended that you create the project's virtual environment with
-  Python 3.8 as some dependencies are not yet available for Python 3.9
+- Go back to the root of this repo (`cd ..`) and run `yarn setup:venv` to set up
+  a Python virtual environment in the demo project (in `demo/.env`).
```
Comment (jorgeorpinel) [inline CONTRIBUTING.md:59]: Does it work if you `yarn setup:venv` first from the root, and then `cd` into `demo/` for everything else? Instead of going back and forth in each step... ⏳

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -47,47 +47,48 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
...
-  setup the Python virtual environment in the demo project.
+- Pull the project data with the extension (**DVC panel** in the _Source Code_
+  view) or with `.env/bin/dvc pull` from command line.
 
-- Once the environment has been setup you should navigate into the demo folder
-  activate the virtual environment and run `dvc pull`.
+- In order to run experiments in the demo project (from the **DVC View**), the
+  virtual env should be loaded. This will happen automatically if the Microsoft
+  [Python extension] is installed and setup to use it. Or you can set it up
+  manually with the `dvc.dvcPath` setting (see [Configuration]).
```
Comment (jorgeorpinel) [inline CONTRIBUTING.md:69]: I'm using the MS Python ext but is the latter true? Can you set `dvc.dvcPath` to `.env/bin/dvc` even if `ms-python.python` is not configured to use `.env/bin/python`? ⏳

Comment (jorgeorpinel) [reply CONTRIBUTING.md]: > Can you set dvc.dvcPath to .env/bin/dvc even if ms-python.python is not configured to use .env/bin/python?

Tried that and it doesn't work (I get this issue: https://github.com/iterative/vscode-dvc/issues/1927). Changing this to require the `ms-python.python` for now... ✔️

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -47,47 +47,50 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
...
-- Once the environment has been setup you should navigate into the demo folder
-  activate the virtual environment and run `dvc pull`.
+- In order to run experiments in the demo project (from the **DVC View**), the
+  virtual env should be loaded through the Microsoft [Python extension] (please
+  install it), having this in `.vscode/settings.json`:
 
-- In order to run experiments in our demo project we require the Python virtual
-  environment to be loaded.
+  ```json
+  "python.defaultInterpreterPath": "demo/.env/bin/dvc"
```
Comment (jorgeorpinel) [inline CONTRIBUTING.md:71]: BTW why don't we just include this in the repo's .vscode/settings.json file, which is committed to Git?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -47,47 +47,48 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
...
 
 <a id='warning'></a>
 
-> **Warning**  
-> When using any project that relies on an isolated Python environment (`conda`,
-> `venv`, etc.), Microsoft's
-> [Python extension](https://github.com/Microsoft/vscode-python) is required.
+> **Warning** : When using any project that relies on an isolated Python
+> environment (`conda`, `venv`, etc.), Microsoft's Python extension is required.
 > It's used by this extension to locate and utilize the required environment.
```
Comment (jorgeorpinel) [inline CONTRIBUTING.md:83]: Why is this the case? Why should some other extension ever be required for this extension to work?

File: CONTRIBUTING.md (hunk 2 above)
Comment (jorgeorpinel) [reply CONTRIBUTING.md]: > There's no activate script.

I went through this again but running `yarn setup:env` as the first step, and there's an `activate` script now 🤷🏼 so this is answered I think. I'll update the instructions... ✔️

Comment (jorgeorpinel) [reply CONTRIBUTING.md]: Still not sure what kind of venv it is from the virtualenv-install.ts script, TBH. I guess readers don't need to know exactly but I'm curious: is it the same as a `python -m venv` ?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -47,47 +47,50 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
...
 
-- Go to the `demo/` directory of this repo.
+- Install [Yarn](https://yarnpkg.com/) if needed. Run `yarn setup:venv` from the
+  root of this repo to set up a virtual Python environment for the demo project
+  (in `demo/.env`).
 
-- It is recommended that you create the project's virtual environment with
-  Python 3.8 as some dependencies are not yet available for Python 3.9
+- Go to the `demo/` directory of (`cd demo`) and run `yarn install` to set up
+  the development environment.
```
Comment (jorgeorpinel) [inline CONTRIBUTING.md:60]: Last Q (I think): Is this really needed to play around with the demo project? Sounds more like a step for the previous section (**Developer environment**).

File: CONTRIBUTING.md (hunk 2 above)
Comment (shcheklein) [reply CONTRIBUTING.md]: It is `python -m venv` (which is similar to `virtualenv`). It a bit suboptimal btw, since `wheel` is not installed - some packages can be slower / unavailable with this setup.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -47,47 +47,50 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
 
 ## The demo project
 
-The [demo project](demo) was initially forked from
-[dvc-checkpoints-mnist](https://github.com/iterative/dvc-checkpoints-mnist/tree/make_checkpoint),
-and is provided as a lightweight, convenient testbed to try the extension out
-with. It is not an exhaustive showcase of DVC's features, so testers are
-encouraged to try other DVC repositories -- especially real-world cases!
+The [demo project](demo) was initially forked from [dvc-checkpoints-mnist], and
```
Comment (shcheklein) [inline CONTRIBUTING.md:50]: let's remove this reference, it doesn't matter at all now I think

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -47,47 +47,50 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
...
+try other DVC repositories -- especially real-world cases!
 
-- Go to the `demo/` directory of this repo.
+- Install [Yarn](https://yarnpkg.com/) if needed. Run `yarn setup:venv` from the
+  root of this repo to set up a virtual Python environment for the demo project
+  (in `demo/.env`).
 
-- It is recommended that you create the project's virtual environment with
-  Python 3.8 as some dependencies are not yet available for Python 3.9
+- Go to the `demo/` directory of (`cd demo`) and run `yarn install` to set up
```
Comment (shcheklein) [inline CONTRIBUTING.md:59]: yes, it's not needed. `demo` itself even doesn't have the `package.json`.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -47,47 +47,50 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
...
 
-- `yarn install` is a prerequisite for automatically setting up the demo
-  project.
+  > Stay in `demo/` if you want to use DVC from command line (terminal) in the
+  > demo project.
 
-- Run `yarn setup:venv` from the monorepo root (not `./demo`) to automatically
-  setup the Python virtual environment in the demo project.
+- Pull the project data with the extension (**DVC panel** in the _Source Code_
+  view) or with `.env/bin/dvc pull` from command line.
```
Comment (shcheklein) [inline CONTRIBUTING.md:66]: active and run was better

File: CONTRIBUTING.md (hunk 6 above)
Comment (shcheklein) [reply CONTRIBUTING.md:85]: this is fine, if you use Python it'll be installed. We need it since we are using its Python-specific APIs - like detecting and activating all these virtual envs

File: CONTRIBUTING.md (hunk 2 above)
Comment (jorgeorpinel) [reply CONTRIBUTING.md]: OK. Should we make an issue to deprecate virtualenv-install.ts or make it into platform-specific shell scripts? TBH I think just mentioning "create a `virtualenv`" with some links in the README should be good enough.

File: CONTRIBUTING.md (hunk 8 above)
Comment (jorgeorpinel) [reply CONTRIBUTING.md]: Agreed. Esp. since https://github.com/iterative/dvc-checkpoints-mnist has no LICENSE (shoult it?)

Removed.

File: CONTRIBUTING.md (hunk 7 above)
Comment (jorgeorpinel) [reply CONTRIBUTING.md]: UPDATE: Removed completely per https://github.com/iterative/vscode-dvc/pull/1928#pullrequestreview-1012834023

File: CONTRIBUTING.md (hunk 10 above)
Comment (jorgeorpinel) [reply CONTRIBUTING.md]: https://github.com/iterative/vscode-dvc/blob/contrib/demo/CONTRIBUTING.md#the-demo-project

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -47,47 +47,48 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
...
 
-- `yarn install` is a prerequisite for automatically setting up the demo
-  project.
+- Pull the project data with the extension (either from the **DVC Tracked**
+  panel in the _File Explorer_ or the **DVC panel** in the _Source Code_ view)
+  or by running `dvc pull` from a VS Code _Terminal_.
 
-- Run `yarn setup:venv` from the monorepo root (not `./demo`) to automatically
-  setup the Python virtual environment in the demo project.
+  > Remember to activate the virtual env first (`source .env/bin/activate`) if
```
Comment (shcheklein) [inline CONTRIBUTING.md:65]: it should be activated automatically if you run Terminal in VS Code

File: CONTRIBUTING.md (hunk 5 above)
Comment (shcheklein) [reply CONTRIBUTING.md:73]: this is not needed if you open `demo` directly

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -47,47 +47,48 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
...
+cases!
 
-- Go to the `demo/` directory of this repo.
+- Install [Yarn](https://yarnpkg.com/) if needed. Run `yarn setup:venv` from the
+  root of this repo to set up a virtual Python environment for the demo project
+  (in `demo/.env`).
 
-- It is recommended that you create the project's virtual environment with
-  Python 3.8 as some dependencies are not yet available for Python 3.9
+- Important: Open the `demo/` folder of this repo directly in VS Code.
```
Comment (shcheklein) [inline CONTRIBUTING.md:59]: hey @iterative/vs-code - do you primarily open it directly when you run the dev version or from the extension root?

File: CONTRIBUTING.md (hunk 11 above)
Comment (jorgeorpinel) [reply CONTRIBUTING.md:65]: "if you're using your system's terminal" 🙂 

Is the note confusing/unnecessary ? I don't mind removing it.

Comment (jorgeorpinel) [reply CONTRIBUTING.md:65]: Clarified further for now ("... system terminal outside of the IDE")

Comment (shcheklein) [reply CONTRIBUTING.md:65]: yes, seems premature and level of details doesn't correlate with the other places in the document

Comment (jorgeorpinel) [reply CONTRIBUTING.md:65]: Removed.

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -47,47 +47,48 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
...
+  or by running `dvc pull` from a VS Code _Terminal_.
 
-- Run `yarn setup:venv` from the monorepo root (not `./demo`) to automatically
-  setup the Python virtual environment in the demo project.
+  > Remember to activate the virtual env first (`source .env/bin/activate`) if
+  > you're using your system's terminal.
 
-- Once the environment has been setup you should navigate into the demo folder
-  activate the virtual environment and run `dvc pull`.
+- In order to run experiments in the demo project (from the **DVC View**), the
```
[the remark this answers is not in the document]
Comment (jorgeorpinel) [reply CONTRIBUTING.md:65]: That's what this is called, I think?

image

[the remark this answers is not in the document]
Comment (jorgeorpinel) [reply CONTRIBUTING.md:65]: per https://code.visualstudio.com/docs/getstarted/userinterface#_views

[the remark this answers is not in the document]
Comment (jorgeorpinel) [reply CONTRIBUTING.md:65]: Called `dvc-views` under `views` in package.json which is slightly confusing (technically it's a single view with multiple panels for the side bar)

[the remark this answers is not in the document]
Comment (jorgeorpinel) [reply CONTRIBUTING.md:65]: We call it `dvc.viewContainer` (under `contributes.walkthroughs`) and file .../walkthrough/view-container.md but I think "container" is a term we came up with. I'm not seeing that in VS Code docs.

This is all irrelevant to the user of course but it may be good to decide on our own terminology. I vote for just "view" since it matches VS Code docs; I'm happy to update all these internals to singular `dvc(-view)` along with a walkthrough copy edit if needed.

[the remark this answers is not in the document]
Comment (shcheklein) [reply CONTRIBUTING.md:65]: How would you call then actual Views (Experiments, Sort By, Filter, etc) ?

Again, it's probably a bigger discussion. not relevant to this PR ... my comment was not about this at all

[the remark this answers is not in the document]
Comment (jorgeorpinel) [reply CONTRIBUTING.md]: Those are _Panels_ in the _Activity Bar_ of the _DVC View_.

Going back in the text in question, I guess you can run experiments both from several places (Experiments panel or table, or command palette) so no need to mention DVC View specifically... Linking from "run experiments" now instead. PTAL

[the remark this answers is not in the document]
Comment (shcheklein) [reply CONTRIBUTING.md]: I don't think they are panels. We can check the API or sources to see who it's called.

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -47,47 +47,47 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
...
-and is provided as a lightweight, convenient testbed to try the extension out
-with. It is not an exhaustive showcase of DVC's features, so testers are
-encouraged to try other DVC repositories -- especially real-world cases!
+The [demo project](demo) is provided as a lightweight, convenient testbed to try
+the extension out with. It is not an exhaustive showcase of DVC's features, so
+testers are encouraged to try other DVC repositories -- especially real-world
+cases!
 
-- Go to the `demo/` directory of this repo.
+- Install [Yarn](https://yarnpkg.com/) if needed. Run `yarn setup:venv` from the
```
Comment (mattseddon) [inline CONTRIBUTING.md:55]: [I] If you are going to go as far as saying install yarn you also need to mention run `yarn install` or `yarn setup:venv` will not work.

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -47,47 +47,47 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
...
-- Run `yarn setup:venv` from the monorepo root (not `./demo`) to automatically
-  setup the Python virtual environment in the demo project.
+- In order to [run experiments] in the demo project, the virtual env should be
+  loaded through the Microsoft [Python extension] (please install it), having
+  this in `.vscode/settings.json`:
 
-- Once the environment has been setup you should navigate into the demo folder
-  activate the virtual environment and run `dvc pull`.
+  ```json
+  "python.defaultInterpreterPath": "demo/.env/bin/dvc"
```
Comment (mattseddon) [inline CONTRIBUTING.md:70]: [I] This is the option for manually specifying a [Python interpreter](https://code.visualstudio.com/docs/python/environments#_manually-specify-an-interpreter) for the Python extension. Why would you point it at the demo project's version of DVC?

File: CONTRIBUTING.md (hunk 14 above)
Comment (mattseddon) [reply CONTRIBUTING.md:55]: We already state to install yarn on line 22

File: CONTRIBUTING.md (hunk 12 above)
Comment (mattseddon) [reply CONTRIBUTING.md:59]: The normal process is to 

1. Open the project's root
2. Start the dev server
3. Hit f5 to start a debug session
4. Open the demo project in the debug session

**Review by mattseddon — CHANGES_REQUESTED** [review]
Need to consider the whole document, not just the demo project section.

File: CONTRIBUTING.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (mattseddon) [reply CONTRIBUTING.md]: https://code.visualstudio.com/api/ux-guidelines/views#view-containers

[the remark this answers is not in the document]
Comment (jorgeorpinel) [reply CONTRIBUTING.md]: I see, thanks for clarifying VS Code UX terminology. And I see that `dvc-views` is also under `contributes .viewsContainers` indeed... So it's a _View Container_ for the _Primary Sidebar_.

Anyway, for now I just removed the reference and linked to the Run Experiments walkthroughs (PTAL).

But should we use official VS Code terminology (from the dev-facing docs) for README and walkthrough text going forward? I'd need to update "DVC View" in a few places now. Or do we want to have our own terminology like "click the **DVC** button in the _Activity Bar_" ?

File: CONTRIBUTING.md (hunk 12 above)
Comment (jorgeorpinel) [reply CONTRIBUTING.md:59]: Thanks. From this I get that you tend to open the demo/ project directly when using it to test changes to the extension (which is `demo`'s main purpose).

> BTW, I'm having [this issue again](https://github.com/iterative/dvc-checkpoints-mnist/issues/20) in the demo/ project when the branch is not `main` -- I assume contributors will be in new branches when doing all this. I'll try to fix it and if not reach out in Slack for now.

File: CONTRIBUTING.md (hunk 14 above)
Comment (jorgeorpinel) [reply CONTRIBUTING.md:55]: Removing "Install yarn" to simplify.

> Personally I'd replace this whole bullet with a more generic instruction but don't want to deal with the implications rn. See https://github.com/iterative/vscode-dvc/pull/1928#discussion_r903058889 above

File: CONTRIBUTING.md (hunk 15 above)
Comment (jorgeorpinel) [reply CONTRIBUTING.md:70]: Typo, should be `python`. Fixing

File: CONTRIBUTING.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (shcheklein) [reply CONTRIBUTING.md]: > I guess there's some discrepancies

Screen Shot 2022-06-24 at 5 55 26 PM

no, there is no discrepancy I think. Most of the things (all?) in this list are Views. It just happened that in certain cases they are the only Views by default in the containers.

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -47,47 +48,47 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
...
 
-- Once the environment has been setup you should navigate into the demo folder
-  activate the virtual environment and run `dvc pull`.
+- In order to [run experiments] in the demo project, the virtual env should be
+  loaded through the Microsoft [Python extension] (please install it), having
+  this in `.vscode/settings.json`:
 
-- In order to run experiments in our demo project we require the Python virtual
-  environment to be loaded.
+  ```json
```
Comment (shcheklein) [inline CONTRIBUTING.md:70]: we probably want to remove this after all?

Comment (jorgeorpinel) [reply CONTRIBUTING.md]: Didn't get why, but it's been removed now.

Comment (shcheklein) [reply CONTRIBUTING.md]: sorry, to clarify. It should be detected automatically when you open the `demo/` project directly in a lot of cases. In some cases it will depend on how DVC is installed, etc. I feel we just can't cover properly different environments with this settings and in a lot of cases, if everything works find it's not needed

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -47,47 +48,42 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
...
+> encouraged to try other DVC repositories -- especially real-world cases!
 
-- It is recommended that you create the project's virtual environment with
-  Python 3.8 as some dependencies are not yet available for Python 3.9
+- Run `yarn setup:venv` from the root of this repo to set up a virtual Python
+  environment for the demo project (in `demo/.env`).
 
-- `yarn install` is a prerequisite for automatically setting up the demo
-  project.
+- Open the `demo/` folder of this repo directly in VS Code.
```
Comment (shcheklein) [inline CONTRIBUTING.md:60]: minor, I think "directly" is not needed here

Comment (shcheklein) [reply CONTRIBUTING.md:60]: open the folder - can be a bit ambiguous (open in the Editor vs Open as a Folder). But it looks like it's VS Code terminology. May be we can do "open the `demo/` project in VS Code"?

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -47,47 +48,42 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
...
 
-- Run `yarn setup:venv` from the monorepo root (not `./demo`) to automatically
-  setup the Python virtual environment in the demo project.
+- Pull the project data with the extension (either from the **DVC Tracked**
```
Comment (shcheklein) [inline CONTRIBUTING.md:62]: `with the extension` -> using the extension ...

File: CONTRIBUTING.md [hunk 19]
```diff
@@ -47,47 +48,42 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
...
+  panel in the _File Explorer_ or the **DVC panel** for _Source Control_) or by
+  running `dvc pull` from a _Terminal_.
 
-- Once the environment has been setup you should navigate into the demo folder
-  activate the virtual environment and run `dvc pull`.
+- In order to [run experiments] in the demo project, the virtual env should be
+  loaded through the Microsoft [Python extension] (please install it).
```
Comment (shcheklein) [inline CONTRIBUTING.md:67]: just [Python extension] ?

Comment (shcheklein) [reply CONTRIBUTING.md:67]: Python extension should be installed so that env can be activated ...

File: CONTRIBUTING.md [hunk 20]
```diff
@@ -47,47 +48,42 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
...
+> **Warning**: When using any project that relies on an isolated Python
+> environment (`conda`, `venv`, etc.), Microsoft's Python extension is required.
 > It's used by this extension to locate and utilize the required environment.
 
 ## React component development with Storybook
 
-Start Storybook with `yarn storybook` in either the monorepo root or the
-`webview` project, and you can develop the React components this plugin uses
-without requiring VS Code as a dev environment.
+Start Storybook with `yarn storybook` in either the monorepo root or in the
```
Comment (shcheklein) [inline CONTRIBUTING.md:82]: monorepo -> the repo root

File: CONTRIBUTING.md (hunk 17 above)
Comment (jorgeorpinel) [reply CONTRIBUTING.md]: Improved in 9f096503. PTAL

File: CONTRIBUTING.md (hunk 18 above)
Comment (jorgeorpinel) [reply CONTRIBUTING.md:66]:
```suggestion
- Pull the project data using the extension (either from the **DVC Tracked**
```

File: CONTRIBUTING.md (hunk 19 above)
Comment (jorgeorpinel) [reply CONTRIBUTING.md:71]:
```suggestion
  loaded through the [Python extension] (please install it).
```

File: CONTRIBUTING.md (hunk 20 above)
Comment (jorgeorpinel) [reply CONTRIBUTING.md:86]:
```suggestion
Start Storybook with `yarn storybook` in either the root of this repo or in the
```

File: CONTRIBUTING.md (hunk 19 above)
Comment (jorgeorpinel) [reply CONTRIBUTING.md]: Updated. PTAL @shcheklein

File: CONTRIBUTING.md [hunk 21]
```diff
@@ -15,79 +15,78 @@ highly appreciated!
...
 
-- Run `Tasks: Run Build Task` (Ctrl+Shift+b) to start the extension and webview
-  development servers (alternatively run `yarn dev-server` from the terminal).
+- Run `Tasks: Run Build Task` (Shift + Ctrl/Cmd + `b`) to start the extension
+  and Webview dev servers (alternatively, run `yarn dev-server` from the
+  _Integrated Terminal_).
 
-  > **Warning**: Having a separate (`.vsix`) version of the extension installed
+  > **Warning**: Having a separate `.vsix` version of this extension installed
   > may cause all kinds of chaos in your development environment.
```
Comment (jorgeorpinel) [inline CONTRIBUTING.md:32]: Hey @mattseddon 👋🏼 what is this warning meant to prevent? `Tasks: Run Build Task` doesn't seem to generate a .vsix file.

Comment (jorgeorpinel) [reply CONTRIBUTING.md:32]: Also, I think it's just `yarn dev` now?

File: CONTRIBUTING.md [hunk 22]
```diff
@@ -15,79 +15,78 @@ highly appreciated!
...
 
 <a id='warning'></a>
 
-> **Warning**  
-> When using any project that relies on an isolated Python environment (`conda`,
-> `venv`, etc.), Microsoft's
-> [Python extension](https://github.com/Microsoft/vscode-python) is required.
+> **Warning**: When using any project that relies on an isolated Python
+> environment (`conda`, `venv`, etc.), Microsoft's Python extension is required.
 > It's used by this extension to locate and utilize the required environment.
```
Comment (jorgeorpinel) [inline CONTRIBUTING.md:81]: I feel like this note should be moved out to the README or some other place since it can apply to any DVC project and not just to the demo/ project or contrib guidelines. Right?

File: CONTRIBUTING.md (hunk 21 above)
Comment (mattseddon) [reply CONTRIBUTING.md]: definitely dev-server: `"dev-server": "yarn turbo run dev --parallel",`

Comment (mattseddon) [reply CONTRIBUTING.md]: If you have the extension installed (either through a `VSIX` or the marketplace) and start a debug session with the `dev-server` running you're going to be very confused as to what is going on.

File: CONTRIBUTING.md [hunk 23]
```diff
@@ -15,79 +15,78 @@ highly appreciated!
...
-extension. The local repository source is built and loaded into the Extension
-Development Host.
+Setting up a dev environment allows contributors to test changes to this
+extension. The source code is built, and the resulting package loaded into a
+special instance of VS Code.
 
-First, ensure that [Yarn](https://yarnpkg.com/) and
-[Visual Studio Code](https://code.visualstudio.com) are installed.
+First, ensure that [Visual Studio Code](https://code.visualstudio.com) and
+[Yarn](https://yarnpkg.com/) are installed.
```
Comment (mattseddon) [inline CONTRIBUTING.md:23]: Mention `yarn install` dependencies?

File: CONTRIBUTING.md [hunk 24]
```diff
@@ -15,79 +15,78 @@ highly appreciated!
...
-- Run `yarn setup:venv` from the monorepo root (not `./demo`) to automatically
-  setup the Python virtual environment in the demo project.
+- Open the `./demo` project in a VS Code window, for example in the _Extension
+  Development Host_ (see previous section).
 
-- Once the environment has been setup you should navigate into the demo folder
-  activate the virtual environment and run `dvc pull`.
+- Pull the project data using this extension (either from the **DVC Tracked**
+  panel in the _File Explorer_ or the **DVC panel** in _Source Control_) or by
+  running `dvc pull` from _Terminal_.
```
Comment (mattseddon) [inline CONTRIBUTING.md:67]: from the _Integrated Terminal_?

File: CONTRIBUTING.md (hunk 22 above)
Comment (mattseddon) [reply CONTRIBUTING.md:81]: Agreed, I would include it in the Development environment section or at least closer to the top of this one. I think it is ok to duplicate

File: CONTRIBUTING.md (hunk 23 above)
Comment (mattseddon) [reply CONTRIBUTING.md:23]: trying to be as explicit as possible

File: CONTRIBUTING.md (hunk 21 above)
Comment (jorgeorpinel) [reply CONTRIBUTING.md:32]: I have the extension installed via marketplace.

Shift + Cmd + `b` seems to use `yarn run dev`:

image

Ends with

```
webpack 5.73.0 compiled successfully in 5388 ms
No issues found.
```

If I instead run yarn `dev-server` from Terminal, the output looks a bit different but the ending is similar:

```
dvc-vscode-webview:dev: webpack 5.73.0 compiled successfully in 5754 ms
dvc-vscode-webview:dev: No issues found.
```

Start Debugging in either case seems to work fine. The dev host disables all installed extensions anyway:

image

Then again the same happens if I don't run any build task before F5 😅 I've no idea what's happening anyway.

Bottom line: are these instructions clear for extension developers?

Comment (jorgeorpinel) [reply CONTRIBUTING.md:32]: Then again the same happens if I don't run any build task before F5 😅 I've no idea what's happening anyway.

Bottom line: are these instructions clear for extension developers?

Comment (jorgeorpinel) [reply CONTRIBUTING.md:32]: > the same happens if I don't run any build task before F5

Ah, after deleting extension/dist F5 fails.

File: CONTRIBUTING.md (hunk 23 above)
Comment (jorgeorpinel) [reply CONTRIBUTING.md:23]: Added `yarn install` (from root).

File: CONTRIBUTING.md (hunk 24 above)
Comment (jorgeorpinel) [reply CONTRIBUTING.md:68]: https://code.visualstudio.com/docs/editor/integrated-terminal

> If you use a vs code terminal at this point, we know you're in the project root.

File: CONTRIBUTING.md [hunk 25]
```diff
@@ -15,79 +15,80 @@ highly appreciated!
...
-  results of the dev servers installed as an extension, with `Start Debugging`
-  (f5).
+- **Start Debugging** (F5) to open the [Extension Development Host], a child
+  instance of VS Code with the local extension build installed.
 
   > **Note**: using the `Run Extension` command when running the debugger will
   > prevent all other extensions from loading into VS Code. This will improve
   > the performance of VS Code but can cause certain DVC commands to fail if the
-  > project uses an isolated python environment (see [this warning](#warning)).
+  > DVC project uses an isolated Python env (see [this warning](#warning)).
```
Comment (jorgeorpinel) [inline CONTRIBUTING.md:41]: Actually, this warning I'm also not getting: What is the `Run Extension` command?

image

Comment (mattseddon) [reply CONTRIBUTING.md:41]: The options are in a different place for the debugger: 
https://user-images.githubusercontent.com/37993418/176607460-3c1be14f-d168-4651-aa9f-c0045cf455d2.mov

File: CONTRIBUTING.md (hunk 21 above)
Comment (mattseddon) [reply CONTRIBUTING.md]: > Start Debugging after building (with or without webview's webpack watch) seems to work fine. The dev host disables all installed extensions anyway:

This ties in with https://github.com/iterative/vscode-dvc/pull/1928#discussion_r910635641

**Review by mattseddon — APPROVED** [review]

**Comment by jorgeorpinel** [comment]
A couple things may need follow-ups but let's merge this so far I think!

- [ ] Warn about py venvs (require Python extension) in other docs - https://github.com/iterative/vscode-dvc/pull/1928#pullrequestreview-1022559645
- [ ] Remove some vs-code dev warnings? - https://github.com/iterative/vscode-dvc/pull/1928
